src/components/ToDoComponent/TodoComponent.tsx

```diff
@@ -1,11 +1,11 @@
-import { Text, View } from 'react-native';
+import { Text, TouchableOpacity, View } from 'react-native';
 import React, { useEffect, useState } from 'react';
 import Card from './Card';
 import data from './Data';
 import { TodoStyles } from './Styles/TodoStyles';
 import Task from './taskType';
 
-const TodoComponent = () => {
+function TodoComponent({ navigationProp }) {
```

Comment:
Can you please find once which is better, passing navigation instance as a prop or getting is from useNavigation hook in this component?

Reply:
I have used navigation as an instance because the parent stack would know the flow of the navigation. I felt passing the navigation instance as a prop would be better rather than getting it from the useNavigation hook.

Reply:
When you pass the navigation instance as a prop, any changes to the navigation state in the parent component can trigger re-renders in the child component, even if the child component doesn't directly depend on the navigation prop. This can lead to unnecessary re-renders and impact performance, especially if the child component is part of a large component tree..

so please avoid prop drilling as much as possible

Reply:
But how will the 'CreatingGoals' will be passed? It's the title of a screen, how will the parent navigator will know about it?

Reply:
This is done

Reply:
because you already define it in parent component, it knows every screen's name.